### ISSUE TITLE
Fatal Error on Ubuntu Gnome 16.04 Desktop

### ISSUE THREAD

**Issue by dyorgio** [dorkbox/SystemTray#21]
JVM crashes if I try to access MenuEntry by click on icon.
I'm receiving same message of [issue #19](https://github.com/dorkbox/SystemTray/issues/19)
and need to put setIcon BEFORE addMenuEntry to not freeze my app.
But now my app don't freeze, it crashs :)

```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f071470ad84, pid=2426, tid=0x00007f06baeca700
#
# JRE version: Java(TM) SE Runtime Environment (8.0_101-b13) (build 1.8.0_101-b13)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.101-b13 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# C  [libpthread.so.0+0x9d84]  pthread_mutex_lock+0x4
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#

---------------  T H R E A D  ---------------

Current thread (0x00007f070c428800):  JavaThread "GTK Native Event Loop" [_thread_in_native, id=2450, stack(0x00007f06badca000,0x00007f06baecb000)]

siginfo: si_signo: 11 (SIGSEGV), si_code: 128 (SI_KERNEL), si_addr: 0x0000000000000000

Registers:
RAX=0x00007f06f6487370, RBX=0x000000000073e6a0, RCX=0x00000000000000d2, RDX=0x00007f06baec5410
RSP=0x00007f06baec5380, RBP=0x00007f06baec53f0, RSI=0x00007f06baec5400, RDI=0x7475426e6970536b
R8 =0x00007f06baec53f0, R9 =0x00007f070c3ad460, R10=0x00007f06f650c447, R11=0x0000000000000065
R12=0x00007f06baec53ec, R13=0x0000000000000000, R14=0x0000000000753e68, R15=0x00007f06baec6980
RIP=0x00007f071470ad84, EFLAGS=0x0000000000010202, CSGSFS=0x0000000000000033, ERR=0x0000000000000000
  TRAPNO=0x000000000000000d

Top of Stack: (sp=0x00007f06baec5380)
0x00007f06baec5380:   212074786574004c 00007f06f64a1bfa
0x00007f06baec5390:   00007f06baec5400 00007f06baec5410
0x00007f06baec53a0:   0000000000000000 00000000006ec150
0x00007f06baec53b0:   00007f06cec8f161 b170dd48cc475d00
0x00007f06baec53c0:   00000000006ec150 00007f06cec8f161
0x00007f06baec53d0:   00007f06cec90974 00007f06f647df86
0x00007f06baec53e0:   00007f0700000002 00007f06cdc78b23
0x00007f06baec53f0:   0000000000752c00 0000000000000992
0x00007f06baec5400:   000000d4000000d2 00007f0600000000
0x00007f06baec5410:   000000dd000000dc 0000000000000000
0x00007f06baec5420:   0000000000000000 b170dd48cc475d00
0x00007f06baec5430:   00000000007538d0 0000000000753904
0x00007f06baec5440:   00000000006ec150 00007f06cec4993f
0x00007f06baec5450:   0000000000807c20 0000000000000001
0x00007f06baec5460:   0000000000752c00 b170dd48cc475d00
0x00007f06baec5470:   00007f06baec55c0 000000000080ef28
0x00007f06baec5480:   00007f06baec54e0 000000000080ef28
0x00007f06baec5490:   0000000000000000 0000000000753e68
0x00007f06baec54a0:   00007f06baec6980 00007f06cec4be60
0x00007f06baec54b0:   000000000080ee00 00007f06cec1d627
0x00007f06baec54c0:   0000000000753e68 0000000000807cb0
0x00007f06baec54d0:   00007f06baec5500 00007f06cebdbcff
0x00007f06baec54e0:   0000000000000004 0000000000000014
0x00007f06baec54f0:   0000000000000000 0000000000000000
0x00007f06baec5500:   0000000000000000 00007f06cdc7843a
0x00007f06baec5510:   0000000000000000 0000000000000000
0x00007f06baec5520:   0000000000000000 00007f06f4e30f38
0x00007f06baec5530:   0000000100000000 b170dd48cc475d00
0x00007f06baec5540:   0000000000753ca0 0000000000807cb0
0x00007f06baec5550:   0000000000000050 00007f06cebdee13
0x00007f06baec5560:   0000000000000018 000000000077f070
0x00007f06baec5570:   0000000000000005 00000004cc475d00 

Instructions: (pc=0x00007f071470ad84)
0x00007f071470ad64:   f0 02 00 00 00 00 00 00 b8 82 00 00 00 e9 d8 f9
0x00007f071470ad74:   ff ff 66 2e 0f 1f 84 00 00 00 00 00 48 83 ec 08
0x00007f071470ad84:   8b 57 10 89 d1 81 e1 7f 01 00 00 90 89 d0 83 e0
0x00007f071470ad94:   7c 0f 85 95 00 00 00 85 c9 49 89 f8 0f 85 9a 00 

Register to memory mapping:

RAX=0x00007f06f6487370: <offset 0x2b370> in /usr/lib/x86_64-linux-gnu/libX11.so.6 at 0x00007f06f645c000
RBX=0x000000000073e6a0 is an unknown value
RCX=0x00000000000000d2 is an unknown value
RDX=0x00007f06baec5410 is pointing into the stack for thread: 0x00007f070c428800
RSP=0x00007f06baec5380 is pointing into the stack for thread: 0x00007f070c428800
RBP=0x00007f06baec53f0 is pointing into the stack for thread: 0x00007f070c428800
RSI=0x00007f06baec5400 is pointing into the stack for thread: 0x00007f070c428800
RDI=0x7475426e6970536b is an unknown value
R8 =0x00007f06baec53f0 is pointing into the stack for thread: 0x00007f070c428800
R9 =0x00007f070c3ad460 is an unknown value
R10=0x00007f06f650c447: <offset 0xb0447> in /usr/lib/x86_64-linux-gnu/libX11.so.6 at 0x00007f06f645c000
R11=0x0000000000000065 is an unknown value
R12=0x00007f06baec53ec is pointing into the stack for thread: 0x00007f070c428800
R13=0x0000000000000000 is an unknown value
R14=0x0000000000753e68 is an unknown value
R15=0x00007f06baec6980 is pointing into the stack for thread: 0x00007f070c428800


Stack: [0x00007f06badca000,0x00007f06baecb000],  sp=0x00007f06baec5380,  free space=1004k
Native frames: (J=compiled Java code, j=interpreted, Vv=VM code, C=native code)
C  [libpthread.so.0+0x9d84]  pthread_mutex_lock+0x4
C  0x0000000000000992

Java frames: (J=compiled Java code, j=interpreted, Vv=VM code)
j  dorkbox.systemTray.linux.jna.Gtk.gtk_main()V+0
j  dorkbox.systemTray.linux.jna.Gtk.access$300()V+0
j  dorkbox.systemTray.linux.jna.Gtk$1.run()V+34
v  ~StubRoutines::call_stub
```

**Comment by dorkbox** [comment]
Got it working ... just fixing some of the different variations

**Comment by dyorgio** [comment]
Cool 👍 
New branch or master?

**Comment by dorkbox** [comment]
I haven't pushed anything yet, I'm still working out how to get the javaFX version working with stock ubuntu (right now, it works with gnome, just not unity)

**Comment by dyorgio** [comment]
If you create a new branch I can test for you :)

**Comment by dorkbox** [comment]
Almost in a working state. It turns out that Ubuntu 16.04 removed the status notification panel (only the app-indicator panel exists now)... so the GTK tray icon doesn't work anymore in... This is an issue because JavaFX required GTK2, and the default app-indicator in Ubuntu 16.04 is GTK3; so it doesn't work unless the fallback appindicator is installed.

I'm looking into alternatives, such as adding it directly using DBus signals (which I really don't want to do, as I have no clue how to do that)

**Comment by dorkbox** [comment]
3.0 beta is released as a "all-in-one" jar, for a variety of different tray types and configurations. I'm starting a new issue to track the progress for this. I definitely will need help testing in different Linux/Mac/Windows configurations (ie: I have no idea if this works in Windows 10, or the latest MacOsX)

I'm closing this issue and tracking everything in a new one. Please use Issue #22 for all related feedback, problems, etc.
